sql/analyzer/generate_index_scans.go

```diff
@@ -0,0 +1,267 @@
+package analyzer
+
+import (
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/expression"
+	"github.com/dolthub/go-mysql-server/sql/plan"
+	"github.com/dolthub/go-mysql-server/sql/transform"
+)
+
+// generateIndexScans attempts to push conditions in filters down to individual tables. Tables that implement
+// sql.FilteredTable will get such conditions applied to them. For conditions that have an index, tables that implement
+// sql.IndexAddressableTable will get an appropriate index lookup applied.
+// TODO(max): filter pushdown should happen as part of join reordering.
+// A memo should be built before reorder with filter exprGroups, with
+// bitmaps tracking table dependencies. The same processes here should
+// be applied there: 1) filter pushdown to table, 2) filter pushdown to
+// join node, 3) range scan indexes applied to non-lookup tables.
+func generateIndexScans(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope, sel RuleSelector) (sql.Node, transform.TreeIdentity, error) {
```

Comment:
pushdownIndexes would be a better name, I think?

Reply:
"pushdown" doesn't feel like the right verb for choosing an index scan. Filters are pushed between relational nodes. Memo exploration can generate indexscan alternatives for tablescans with filters.  Costing can choose indexscans instead of tablescans.